homeassistant/util/logging.py

```diff
@@ -164,3 +164,46 @@ def wrapper(*args: Any) -> None:
                 log_exception(*args)
         wrapper_func = wrapper
     return wrapper_func
+
+
+def catch_log_coro_exception(
+        func: Coroutine[Any, Any, Any],
```

Comment:
A function is not a coroutine. A function is a Callable.

Reply:
Yes, that was a very unsuitable variable name, fixed!